src/compas_ags/rhino/forceobject.py

```diff
@@ -32,6 +39,122 @@ def inspector_on(self, form):
     def inspector_off(self):
         self.inspector.disable()
 
+    def scale_from_3_points(self):
```

Comment:
ummm.. I thought scale functions should all have nothing to do with anchor... or maybe I think too much...

Reply:
what i said was that it should be at least possible to identify and place the anchor independent of the scaling process. however, if you do 3 point scaling, the expectation of the user will be that the first point is the origin of the scaling process (which is why i suggested the 4 point version since this can be seen more independent of the location of the first point). the way i understand the implementation now is that the scaling will still occur wrt to the existing anchor. this can lead to very unexpected results. perhaps in the 3point version the first point chosen by the user should always be a vertex of the diagram which is then implicitly set as the new anchor...

Reply:
yep... I understand what you mean. I would change. 
(now scale_from_3_points and scale_from_4_points are basically the same, with scale_from_3_points sharing the same start point of reference line 1 and reference line 2... I thought the user would anyway choose the anchor point first for all scale cmds, if they choose the same anchor point as the base then there's no confusion... )

Reply:
or maybe `scale_from_3_points` would automatically pick anchor as the base point, user just need to choose ref pt1 and ref pt2. then in the toolbar, it would be always choose anchor -> scale.

Reply:
sure but then it would be "scale from 2 points" :)

Reply:
ok!